File: python/typelayer_common/typelayer_common/application.py
from . import documentation as typelayer_common_documentation
from . import dynamic as typelayer_common_dynamic
from . import types as typelayer_common_types
from dataclasses import dataclass
from typing import List, Dict, Optional


class ServiceNotFound(typelayer_common_types.TypelayerException):
    requested_service: str;
    available_services: str;
    def __init__(self, m: str, requested_service: str, available_services: str):
        super().__init__(m, "typelayer_common_application", "service_not_found", {"requested_service": requested_service, "available_services": available_services})
        self.requested_service = requested_service;
        self.available_services = available_services;

class FunctionNotFound(typelayer_common_types.TypelayerException):
    requested_service: str;
    requested_function: str;
    available_functions: str;
    def __init__(self, m: str, requested_service: str, requested_function: str, available_functions: str):
        super().__init__(m, "typelayer_common_application", "function_not_found", {"requested_service": requested_service, "requested_function": requested_function, "available_functions": available_functions})
        self.requested_service = requested_service;
        self.requested_function = requested_function;
        self.available_functions = available_functions;

class Process:
    def process_describe(self, ) -> typelayer_common_documentation.ApplicationDescription:
        raise NotImplementedError
    def process_handle(self, encoding: typelayer_common_dynamic.Encoding, path: List[str], body: bytes) -> bytes:
        raise NotImplementedError

File: typescript/typelayer-common/application.ts
// Module: application
import * as typelayerCommonDocumentation from "./documentation";
import * as typelayerCommonDynamic from "./dynamic";
import * as typelayerCommonTypes from "./types";


export class ServiceNotFound extends  typelayerCommonTypes.TypelayerException {
    requestedService: string;
    availableServices: string;
    constructor(m: string, meta: {requestedService: string; availableServices: string;}) {
        super(m, "typelayer_common_application", "service_not_found", new Map([["requested_service", meta.requestedService], ["available_services", meta.availableServices]]));
        Object.setPrototypeOf(this, ServiceNotFound.prototype);
        this.requestedService = meta.requestedService;
        this.availableServices = meta.availableServices;
    }
}

export class FunctionNotFound extends  typelayerCommonTypes.TypelayerException {
    requestedService: string;
    requestedFunction: string;
    availableFunctions: string;
    constructor(m: string, meta: {requestedService: string; requestedFunction: string; availableFunctions: string;}) {
        super(m, "typelayer_common_application", "function_not_found", new Map([["requested_service", meta.requestedService], ["requested_function", meta.requestedFunction], ["available_functions", meta.availableFunctions]]));
        Object.setPrototypeOf(this, FunctionNotFound.prototype);
        this.requestedService = meta.requestedService;
        this.requestedFunction = meta.requestedFunction;
        this.availableFunctions = meta.availableFunctions;
    }
}

export interface Process {
    processDescribe: () => typelayerCommonDocumentation.ApplicationDescription;
    processHandle: (encoding: typelayerCommonDynamic.Encoding, path: Array<string>, body: Uint8Array) => Promise<Uint8Array>;
}